### PR TITLE
Add edit meal feature

### PR DESCRIPTION
**Pull request by qadriyyah119**

### CODE REVIEW

File: TrackMyMeal/Controllers/TrackMyMealTableViewController.swift [hunk 1]
```diff
@@ -138,20 +141,61 @@ class TrackMyMealTableViewController: UITableViewController, AddNewMealViewDeleg
...
+            let currentMeals = mealList.mealSectionList(for: sectionCategory)
+            if let index = currentMeals.firstIndex(where: { $0.id == oldMealListItem.id }) {
+              let oldIndexPath = IndexPath(row: index, section: oldMealSectionIndex)
+              mealList.meals.remove(at: index)
+              mealList.meals.append(meal)
+              
+              tableView.deleteRows(at: [oldIndexPath], with: .automatic)
+            }
+            tableView.beginUpdates()
+            tableView.reloadSections(IndexSet(integer: oldMealSectionIndex), with: .automatic)
```
Comment (Marquis103) [inline TrackMyMeal/Controllers/TrackMyMealTableViewController.swift]: do something like `IndexSet(arrayLiteral: [oldMealSectionIndex, newMealSectionIndex])`

File: TrackMyMeal/Controllers/TrackMyMealTableViewController.swift [hunk 2]
```diff
@@ -138,20 +141,61 @@ class TrackMyMealTableViewController: UITableViewController, AddNewMealViewDeleg
...
+  func addNewMealViewController(_ controller: AddNewMealViewController, didFinishEditing meal: MealListItem, oldMealListItem: MealListItem ,oldMealSectionIndex: Int, newMealSectionIndex: Int) {
+    
+    for sectionCategory in MealCategory.allCases {
+      let currentMeals = mealList.mealSectionList(for: sectionCategory)
+      if let index = currentMeals.firstIndex(where: { $0.id == meal.id }) {
+        let indexPath = IndexPath(row: index, section: newMealSectionIndex)
+        
+        if oldMealSectionIndex == newMealSectionIndex {
+          tableView.beginUpdates()
+          tableView.reloadRows(at: [indexPath], with: .automatic)
```
Comment (Marquis103) [inline TrackMyMeal/Controllers/TrackMyMealTableViewController.swift]: remove reloadRows

File: TrackMyMeal/Controllers/TrackMyMealTableViewController.swift (hunk 1 above)
Comment (qadriyyah119) [reply TrackMyMeal/Controllers/TrackMyMealTableViewController.swift]: Done - tableView.reloadSections(IndexSet(arrayLiteral: oldMealSectionIndex, newMealSectionIndex), with: .automatic)

File: TrackMyMeal/Controllers/TrackMyMealTableViewController.swift (hunk 2 above)
Comment (qadriyyah119) [reply TrackMyMeal/Controllers/TrackMyMealTableViewController.swift]: Done

**Pull request merged**